### PR TITLE
Start sectioning off ReactiveSwift

### PR DESCRIPTION
**Pull request by lickel**
I want to wean the core off of ReactiveSwift. 

This PR does two things:
* Update Package.swift
    * Reorganize the file
    * Expose the RxSwift flavors
    * Update ReactiveSwift to be consistent with the Podspec
* Remove `import ReactiveSwift` for anything that didn't really need it

## Checklist

- [ ] Unit Tests
- [ ] UI Tests
- [ ] Snapshot Tests (iOS only)
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: WorkflowReactiveSwift/Sources/QueueScheduler+Workflow.swift [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2020 Square Inc.
```
Comment (dnkoutso) [inline WorkflowReactiveSwift/Sources/QueueScheduler+Workflow.swift:2]: nit of nits: 2023?

**Review by dnkoutso — APPROVED** [review]

File: Package.swift [hunk 2]
```diff
@@ -22,26 +31,42 @@ let package = Package(
...
+            name: "WorkflowReactiveSwiftTesting",
+            targets: ["WorkflowReactiveSwiftTesting"]
         ),
+
+        // MARK: WorkflowRxSwift
+
         .library(
-            name: "WorkflowTesting",
-            targets: ["WorkflowTesting"]
+            name: "WorkflowRxSwift",
```
Comment (n8chur) [inline Package.swift:49]: Should this be `WorkflowReactiveSwift`?

Comment (lickel) [reply Package.swift:49]: Whoops! it was the `target` that was the issue

File: Workflow/Sources/DispatchQueue+Workflow.swift [hunk 3]
```diff
@@ -15,16 +15,8 @@
...
-}
-
-extension QueueScheduler {
-    static let workflowExecution: QueueScheduler = QueueScheduler(
-        qos: .userInteractive,
-        name: "com.squareup.workflow",
-        targeting: DispatchQueue.workflowExecution
-    )
+    @_spi(WorkflowInternals)
+    public static let workflowExecution: DispatchQueue = .main
```
Comment (jamieQ) [inline Workflow/Sources/DispatchQueue+Workflow.swift:21]: if it's now public, is the SPI needed?

Comment (kyleve) [reply Workflow/Sources/DispatchQueue+Workflow.swift:21]: Yep – you can _only_ SPI public API

**Pull request closed without merging**